File: 14-Longest-Common-Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String stable =strs[0];
            for(int word =1; word < strs.length; word++){
                while(!strs[word].startsWith(stable)){
                    stable = stable.substring(0,stable.length() -1);
                    if(stable.isEmpty()) return \\;
                }
            }
        return stable;
    }
}

File: 14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        prefix = strs[0]

        for word in range(1, len(strs)):
            while prefix != strs[word][:len(prefix)]:
                prefix = prefix[:-1]
                if prefix== \\:
                    return \\
        return prefix

            
